### PR TITLE
fix: Allow tabbing out of the title

### PR DESCRIPTION
**Pull request by acezard**
https://trello.com/c/cz5BooES/98-%F0%9F%93%9D-notes-pouvoir-sortir-du-titre-apr%C3%A8s-un-entr%C3%A9e

### CODE REVIEW

**Review by Crash-- — CHANGES_REQUESTED** [review]
see comment

**Comment by acezard** [comment]
> `with-outer-listeners` is only used on `DropDown` stuff. We can't rely on this to provide this kind of side effect. If one day we decide to remove this DropDown UI, we'll break this kind of stuff.
> 
> We already have logic to handle the title attribute. We should add this logic there https://github.com/cozy/cozy-notes/blob/master/src/components/notes/editor-view.jsx#L64-L66

I agree my implementation is most hacky. But we can't use onTitleBlur, we never trigger blur in the current implementation. Please have a call to talk about this

**Comment by Crash--** [comment]
Let's create a new listener for the textarea. But not in `with-outer-listeners`. We should add it to `EditorView`

**Comment by acezard** [comment]
useTitleChanges

File: src/hooks/useTitleChanges.js [hunk 1]
```diff
@@ -6,6 +6,10 @@ function useTitleChanges({ noteId, title, setTitle, serviceClient }) {
   const onLocalTitleChange = useCallback(
     serviceClient
       ? e => {
+          // Let the user tab out of the title when using the 'Enter' key
+          if (e.nativeEvent?.inputType === 'insertLineBreak')
+            return window.document.querySelector('.ProseMirror').focus()
```
Comment (Crash--) [inline src/hooks/useTitleChanges.js:11]: This is not what we asked. We want a "tab", not a focus on this selector

Comment (acezard) [reply src/hooks/useTitleChanges.js:11]: But there is no API to ask a tab event to the browser. The best we can do is find the next focusable element (tabindex), but some focusable elements do not have tabindex. What would you suggest?

Comment (Crash--) [reply src/hooks/useTitleChanges.js:11]: https://developer.mozilla.org/fr/docs/Web/API/EventTarget/dispatchEvent
```
window.dispatchEvent(new KeyboardEvent('keydown', {
  'key': 9
}));
```

will work, no?

Comment (acezard) [reply src/hooks/useTitleChanges.js:11]: It does, that's smart

Comment (acezard) [reply src/hooks/useTitleChanges.js]: @Crash-- updated

**Review by Crash-- — APPROVED** [review]

**Pull request merged**